### ISSUE TITLE
Cannot get subscriptions 

### ISSUE THREAD

**Issue by dsiq-rd** [python-amazon-mws/python-amazon-mws#176]
I have done everything including creating a new SQS queue, registering the queue and creating subscriptions. When I list registered urls and subscriptions I see everything as expected but I am not able to receive messages.

My queue is also having the required permissions and pricinipal set as per documentation.

I would appreciate any help on this.

**Comment by elcolumbio** [comment]
With the information you provided, my first guess it is a boto3 issue. Assuming you use boto3. The docs are here https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html

**Comment by dsiq-rd** [comment]
Thank you @elcolumbio . I don't see it as Boto3 Issue. I have actually created the SQS Queue through this script.

```
def createNewQueue(self, QueueName):
		sqsResource = boto3.resource('sqs')
		sqs_helper = sqsQueueHelper()
		allQueues = sqs_helper.getallQueues()
		queueNames=[]
		for url in allQueues:
			url = url.split('/')
			queueNames.append(url[-1])
		if QueueName not in queueNames:
			try:
				createQueue = sqsResource.create_queue(
					QueueName=QueueName, 
					Attributes={'DelaySeconds':'5'}
					)
				addPerm = sqs_helper.addPermissions(createQueue.url)
				print(createQueue.url)
				return createQueue.url
			except:
				raise
		else:
			return jsonify(message="The client queue already exist"), 409
```

and then I have added permissions through this:

```
	def addPermissions(self, QueueUrl):
		principal = '437568002678' #Amazon Pricinpal to be set for receiving messages
		sqsClient = boto3.client('sqs')
		addPermission = sqsClient.add_permission(
			QueueUrl=QueueUrl,
			Label='testing',
			AWSAccountIds=[principal],
			Actions = ['SendMessage','GetQueueAttributes']
			)
		print(addPermission)
		return addPermission
```

After the Queue is created, Step 1: this is how I have first registered the Queue URL:
```
def registerSQSEndpoint(self, kwargs):
		#Notificaiton Types that can be registered
		NotificationType = ['ReportProcessingFinished','FeedProcessingFinished']
		
		#Create MWS Main Object
		mwsObj = subs.Subscriptions(kwargs['access_key'],
			kwargs['secret_key'],
			kwargs['account_id'],
			region=kwargs['region']
			auth_token=kwargs['mws_auth_token'])

# Register Queue URL Step 1
		registerUrl=mwsObj.register_destination(kwargs['marketplace_id'],{'sqsQueueUrl':kwargs['sqsQueueUrl']})
		print(registerUrl.original)
		time.sleep(5)
		
# Create Subscriptions - Step 2
		for types in NotificationType:
			response = mwsObj.create_subscription(kwargs['marketplace_id'],{'sqsQueueUrl':kwargs['sqsQueueUrl']},notification_type=types)
			print(response.original)
			time.sleep(5)
```

When I get the list of destination and the list of subscription, everything seems to be showing as expected. But apparently for no reason the notifications doesn't come through!

Has anyone faced this issue, or am I doing something wrong here?

**Comment by Bobspadger** [comment]
have you checked the permissions for amazon are showing on the control panel for the queue?
have you made sure its NOT a fifo queue?

**Comment by elcolumbio** [comment]
Also you can test with this function.
test = sub.send_test_notification_to_destination(marketplace_id, attributes, "SQS")

**Comment by Bobspadger** [comment]
this is my very basic setup code for getting a subscription running.
```python
# flake8: noqa
import mws

from credentials import creds
from helpers.mws_factory import MWSFactory

MARKETPLACE_ID = creds['amazon']['marketplaceids']['UK']
SHIPPING_PRICE = 4.50

subscriptions = MWSFactory.factory('subscriptions')

att_list = {"sqsQueueUrl": "https://sqs.eu-west-2.amazonaws.com/378051892504/MWS_DEV"}


## Register a destination
try:
    new_sub = subscriptions.register_destination(MARKETPLACE_ID, delivery_channel='SQS', attributes=att_list )
    print(new_sub)
except Exception as e:
    print(e)

# Create a subscription
try:
    create_subscription = subscriptions.create_subscription(marketplace_id=MARKETPLACE_ID, delivery_channel='SQS', attributes=att_list, notification_type='AnyOfferChanged', is_enabled=True )
    print(create_subscription)
except Exception as e:
    print(e)

# Check the subcriptions to make sure its taken
try:
    get_subscriptions = subscriptions.get_subscription(marketplace_id=MARKETPLACE_ID, delivery_channel='SQS', attributes=att_list, notification_type='AnyOfferChanged')
    print(get_subscriptions)
except Exception as e:
    print(e)


# try:
#     dereg = subscriptions.deregister_destination(marketplace_id=MARKETPLACE_ID, attribute_list=att_list)
#     print(dereg)
# except Exception as e:
#     print(e)

send_test_notification = subscriptions.send_test_notification_to_destination(marketplace_id=MARKETPLACE_ID, delivery_channel='SQS', attributes=att_list)
print(send_test_notification)

subs = subscriptions.list_subscriptions(marketplace_id=MARKETPLACE_ID)
destination = subscriptions.list_registered_destinations(marketplace_id=MARKETPLACE_ID)
print(subs)
```

please note:

1. I've created the queue manually on amazon aws 
2. I've applied the access policies as per the documentation via the web console.
3. I'm then using the connection strings provided in there to run the rest of the code.

**Comment by Bobspadger** [comment]
also, I'm calling the subscriptions api from mws via my own factory system to carry out the authentication outside of the code (do it once, use everywhere)

**Comment by dsiq-rd** [comment]
@Bobspadger This is awesome! Thanks for sharing this. I haven't used Factory until now. Let me read the docs and implement it... Will try and let you know. It seems pretty straight forward though.

**Comment by dsiq-rd** [comment]
> have you checked the permissions for amazon are showing on the control panel for the queue?
> have you made sure its NOT a fifo queue?

Yes sir, the permissions are set accurately, and they are standard queue.

**Comment by dsiq-rd** [comment]
Finally, I figure it out!!! This was an Amazon side issue and not the other way around. It drove me crazy for a while. And tonight I am getting a throttling error and rate exceeded errors on first request. When I looked the API status on their website, it said that they are seeing errors, but the service was Green.

Anyways, one of my queue is working and not the new ones that I just created. Also, this is an issue across Europe today. Hope it gets better tomorrow. Thank you again for everyone that responded.
